lib/ice_cube/parsers/ical_parser.rb

```diff
@@ -5,17 +5,19 @@ def self.schedule_from_ical(ical_string, options = {})
       ical_string.each_line do |line|
         (property, value) = line.split(':')
         (property, tzid) = property.split(';')
+        (_, time_zone) = tzid.split('=') if tzid
```

Comment:
I believe this won't always work - looking at the spec, a component may have many parameters, of the form `<KEY>=<VALUE>`, separated by semicolons. If the key is `TZID`, then the value is the time zone that you want. But it there are other keys too.

For example, this is a valid ical line: `DTSTART;VALUE=DATE:20200525`. The code here would parse this and get the time zone `DATE`, which is not correct. The `VALUE` parameter tells us that the value (after the colon - `20200525` should be parsed as a date.

Another example would be `DTSTART;VALUE=DATE-TIME;TZID:US-Eastern:19980119T020000`. Here we do have a time zone, but it would be discarded by the code we have, and instead the time zone would be `DATE-TIME`.

Reply:
@iainbeeston you're correct.

This might fix it:

```ruby
(property, *params) = property.split(';')
params.map! { |p| p.split('=') }
time_zone = params.select { |p| p&.first&.upcase == "TZID" }.first&.last
value_type = params.select { |p| p&.first&.upcase == "VALUE" }.first&.last
```

`TimeUtil.deserialize_time` should also be updated so that the `Hash` case accepts a `type` key and handles both `"DATE"` and `"DATE-TIME"` cases (I don't think the `"TIME"` value is applicable to this library).